src/main/java/com/emc/object/s3/jersey/S3JerseyClient.java

```diff
@@ -527,7 +527,11 @@ public PutObjectResult putObject(PutObjectRequest request) {
 
         // enable checksum of the object
         request.property(RestUtil.PROPERTY_VERIFY_WRITE_CHECKSUM, Boolean.TRUE);
-
+        Object object = request.getObject();
+        if (object instanceof String) {
```

Comment:
Which test case hit this http 411 error? A quick search shows that 411 error response code indicates that the server refuses to accept the request without a defined Content-Length header. I feel there might be better place for the change.